### ISSUE TITLE
[Feature request] better error messages when types mismatch

### ISSUE THREAD

**Issue by fmassa** [rbgirshick/yacs#1]
If the yaml file containing the configuration options has a type mismatch (e.g., an `int` instead of a `float`), the error message is currently not very helpful and it is hard to spot where the error is.

For example
```
  File "test_net.py", line 57, in main
    cfg.merge_from_file(args.config_file)
  File "/private/home/fmassa/.conda/envs/detectron/lib/python3.6/site-packages/yacs/config.py", line 113, in merge_from_file
    cfg = CfgNode(load_cfg(f))
  File "/private/home/fmassa/.conda/envs/detectron/lib/python3.6/site-packages/yacs/config.py", line 226, in load_cfg
    return _to_cfg_node(cfg_as_dict)
  File "/private/home/fmassa/.conda/envs/detectron/lib/python3.6/site-packages/yacs/config.py", line 258, in _to_cfg_node
    return convert_to_cfg_node(cfg_dict)
  File "/private/home/fmassa/.conda/envs/detectron/lib/python3.6/site-packages/yacs/config.py", line 255, in convert_to_cfg_node
    cfg_node[k] = convert_to_cfg_node(v)
  File "/private/home/fmassa/.conda/envs/detectron/lib/python3.6/site-packages/yacs/config.py", line 250, in convert_to_cfg_node
    assert _valid_type(cfg_dict)
AssertionError
```

Ideally, it would be good to be explicit on which argument caused the problem, what was its type, and what was the expected type.

----
EDIT: the error was actually due to a misformated yaml, but the error message was not really very helpful in this case.

**Comment by rbgirshick** [comment]
@fmassa great suggestion. In general, I try to make the `yacs` error messages actionable, but I missed this one. Can you provide a repro of your case so I can make sure that the message is meaningful in the situation you encountered (sending by DM is fine).

**Comment by rbgirshick** [comment]
This should be addressed by #2.

**Comment by fmassa** [comment]
Fixed via #2
